tests/test_pipeline/test_base.py

```diff
@@ -8,9 +8,392 @@
 
 from etna.datasets import TSDataset
 from etna.distributions import BaseDistribution
+from etna.pipeline import FoldMask
 from etna.pipeline.base import BasePipeline
 
 
+@pytest.mark.parametrize(
+    "first_train_timestamp, last_train_timestamp, target_timestamps",
+    [
+        (None, pd.Timestamp("2020-01-05"), [pd.Timestamp("2020-01-06")]),
+        (None, pd.Timestamp("2020-01-05"), ["2020-01-06"]),
+        (None, "2020-01-05", [pd.Timestamp("2020-01-06")]),
+        (None, "2020-01-05", ["2020-01-06"]),
+        (None, "2020-01-05", ["2020-01-06", "2020-01-07"]),
+        (None, "2020-01-05", ["2020-01-06", "2020-01-06"]),
```

Comment:
Why it is a correct case?

Reply:
Because under the hood in `FoldMask` I added deduplication of `target_timestamps`. 

Earlier there was only sorting, I thought that it makes sense to deduplicate. I'm not sure is it a good decision or not.

Reply:
I am not sure it is a good decision as some one can do it by mistake and every thing will work fine -- not sure that it is the best behaviour is such case

Reply:
What do you think we should do if there is a duplication in `target_timestamps`?

Reply:
Fail -- we can consider the old behaviour as bug

Reply:
I reverted the old behavior with a duplication: it just sorts `target_timestamps` and does nothing with duplication.